### PR TITLE
[docs] Capture `power()` in docs/function/numeric

### LINKED ISSUE #1082: Update - Capture `power()` function
Describe the requested changes:

In the core repo QuestDB has `power` function that is not yet documented:

[PowDoubleFunctionFactory.java](https://github.com/questdb/questdb/blob/master/core/src/main/java/io/questdb/griffin/engine/functions/math/PowDoubleFunctionFactory.java)

The task involves:
Create a new section for the function, following the [template](https://github.com/questdb/questdb.io/blob/master/docs/__guidelines/template/function.md).

Location:
[/docs/reference/operators/rege/](https://questdb.io/docs/reference/function/numeric/)

### PR DESCRIPTION
**Pull request by sudiptog81**
Adds documentation for `power()` function along with relevant examples in the relevant section. 
Tested on local development setup.

Closes #1082.

image

### CODE REVIEW

File: docs/reference/function/numeric.md [hunk 1]
```diff
@@ -60,6 +60,30 @@ SELECT log(4.123)
 | ------------ |
 | 1.4165810537 |
 
+## power
+
+`power(base, exponent)` returns the value of a number `base` raised to the
+power of some number `exponent`.
```
Comment (amyshwang) [inline docs/reference/function/numeric.md:66]:
```suggestion
power defined by `exponent`.
```

**Review by amyshwang — CHANGES_REQUESTED** [review]
Minor editorial comment.

File: docs/reference/function/numeric.md [hunk 2]
```diff
@@ -60,6 +60,30 @@ SELECT log(4.123)
...
+- `exponent` is any numeric value.
+
+**Return value:**
+
+Return value type is `double`.
+
+**Examples:**
+
+```questdb-sql
+SELECT power(2, 3)
```
Comment (puzpuzpuz) [inline docs/reference/function/numeric.md:80]: nit:
```suggestion
SELECT power(2, 3);
```

**Review by puzpuzpuz — APPROVED** [review]
@sudiptog81 thanks for the contribution!

**Pull request merged**